### ISSUE TITLE
Issue in setting random seed while using NeuralProphet

### ISSUE THREAD

**Issue by atanuray099** [ourownstory/neural_prophet#631]
**While fitting the NeuralProphet model, we are getting different results each time, even if we are setting the seed :** 

from neuralprophet import set_random_seed
import torch
import os

set_random_seed(89)
seed = 89
random.seed(seed)
os.environ['PYTHONHASHSEED'] = str(seed)
np.random.seed(seed)
torch.manual_seed(seed)

Any particular reason ? 
Looking forward to your support .

**Comment by bhausleitner** [comment]
Hi @atanuray099, 

thank you so much for reaching out and creating a ticket. First, the function `np.random.seed()` is outdated and shouldn't be used anymore (https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html).

The variability of results stems from stochastic gradient descend trying to find different optima, the random initialization of weights as well as different learning rates and different shuffling of the dataloader.

You're absolutely correct, set_random_seed should prevent this and lead to identical results.

Can you please share the output of your function? Did you plot the results and printed the exact same numbers?

**Comment by bhausleitner** [comment]
Here is the function: 
<img width="371" alt="image" src="https://user-images.githubusercontent.com/58265021/166073325-540ccf79-7fdf-4d04-a42d-9608a2fdb079.png">

**Comment by jeeva-98** [comment]
https://colab.research.google.com/drive/1AcmmsoiDUzAaZcTmuS8wtgiodjB_sKXL#scrollTo=uylxKojbFFSx

pls review this colab notebook

i have setted the seed properly

still it is giving different results while finding the test metrics

**Comment by bhausleitner** [comment]
Hi @jeeva-98, 

Thanks for providing such a detailed description of this behavior. 

I have checked your repo and ran your code locally. And I experience different results as well. I still think this has something to do with numpy's library, did you try the same function for other ML models? 

Thanks!

**Comment by jeeva-98** [comment]
Yes..Can you give a new update on neural prophet solving this issue. Looking forward to the support

thanks

**Comment by bhausleitner** [comment]
Hi @jeeva-98, 

thanks, we will do!

**Comment by ourownstory** [comment]
Closing due to inactivity. If more people have this issue, we will reopen the issue.
